packages/amesos2/src/Amesos2_Tacho_def.hpp

```diff
@@ -109,28 +112,20 @@ TachoSolver<Matrix,Vector>::symbolicFactorization_impl()
         TEUCHOS_TEST_FOR_EXCEPTION(sp_colind == nullptr,
           std::runtime_error, "Amesos2 Runtime Error: sp_colind returned null");
 
-      // Tacho uses size_type size_t which matches Tpetra but not Epetra (int)
-      // So we need a converter
+      // Tacho size_type is int or size_t based on -DTacho_ENABLE_INT_INT
       row_ptr = size_type_array("r", this->globalNumRows_ + 1);
       for(global_size_type n = 0; n < this->globalNumRows_ + 1; ++n) {
         row_ptr(n) = static_cast<size_type>(sp_rowptr[n]);
       }
 
-      // Now convert the cols - now on an earlier version I thought this was
-      // working as a straight conversion but then some things have changed
-      // and it seems we will need to convert. TODO: Assess this conversion.
-      // Can we do this more optimally and still work for all Epetra/Tpetra
-      // cases.
+      // Tacho oridinal_type is int
       cols = ordinal_type_array("c", this->globalNumNonZeros_);
       for(global_size_type n = 0; n < this->globalNumNonZeros_; ++n) {
         cols(n) = static_cast<ordinal_type>(sp_colind[n]);
       }
-
-      // TODO - For Tpetra, we could have a direct view like this...
-      // Can we explot this here?
-      //row_ptr = size_type_array(sp_rowptr, this->globalNumRows_ + 1);
     }
     else
+*/
     {
```

Comment:
Overall, we can do both. 
```
if (std::is_same<size_type,typename decltyp(sp_rowptr)::non_const_value_type>::value) {
  // use view directly via shallow copy
} else {
  // convert
}
```

Reply:
If we use the direct shallow copy we still need that code to compile for the non optimized cases. I have updated with a working method but have to use a force cast to achieve this. I'm not sure if this is the best way to handle this situation. I've tested the current in SEMS and seems ok for complex on and off.